packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -2664,12 +2664,19 @@ class BuildOwner {
   }
 
   final Map<GlobalKey, Element> _globalKeyRegistry = <GlobalKey, Element>{};
-  final Set<Element> _debugIllFatedElements = HashSet<Element>();
+
+  // In Profile/Release mode this field is initialized to `null`. The Dart compiler can
+  // eliminate unused fields, but not their initializers.
+  final Set<Element>? _debugIllFatedElements = kDebugMode ? HashSet<Element>() : null;
```

Comment:
Interesting. Can we come up with some kind of a check to avoid re-introducing debug only fields that are not initialized to null in release mode? A lint is probably a little to specific?

Reply:
If you want to maintain it, I could write a regex test... :)

Reply:
I think a regex test is probably the best we can do right now. We should just add comments explaining what it does and why.

Reply:
I started writing a regex for this last week and thoguht we'd probably have to do some kind of post filtering on it to avoid it flagging things that are trivial to initialize like bools or ints.

Reply:
Maybe analyzer package is a better way to do it.

Reply:
@goderbauer WDYT of just having this check apply to any field that has `debug` in the name? 

I think people will forget to attach this annotation to new debug fields, and I think we're probably missing similar cases in other core classes in the framework.

Reply:
I'm still fine with landing this change as is though, it's a clear improvement

Reply:
I am leaning towards opt in because it is not impactful on fields initialized to constants

Reply:
Yeah, ideally we would apply this to all fields with debug in the name. The analysis should be able to detect that something is initialized to a const value and not complain? (At least for the likely simple cases).

At some point we could/should extent the analysis to ensure that debug only fields are only used from asserts and other debugOnly methods to ensure they will actually be shaken out.

Having this opt-in for now is fine, too, though.

Reply:
I think super ideally the compiler would just eliminate both the field and the initializer and we could delete the test :)

Reply:
lol, yes.

Reply:
The compiler would have to understand the difference between side effects you want and side effects you don't.  But maybe this could be special cased.. It's just a weird perf cliff :\

You could imagine a type that actually does work in its initializer though, and users depend (perhaps unintentionally) on that work getting done.

Reply:
The bugs @mraleph has filed related to this would be helpful